### ISSUE TITLE
[9.5.3] When ticket created by mailgate, occasionally assigns a already used Ticket ID, and does not actually create ticket or modify the existing one it takes the ID from.

### ISSUE THREAD

**Issue by mrjd19811** [glpi-project/glpi#8448]
**Describe the bug**

When ticket created by mailgate, it will occasionally assign a already used Ticket ID, and does not actually create a new ticket or modify the existing one it takes the ID from.

**Page(s) URL**

n/a

**To reproduce**

Send new ticket in by e-mail.

**Expected behavior**

Create new ticket if unrelated to old.
or Append to old ticket if related.

**Logs**
Could not find any relevant logs even at max level. No logs are written when it fails it appears. The only notice we get is our teams relay. Which I will show in screenshot.

**Screenshots**
![image](https://user-images.githubusercontent.com/71107353/103031558-ae936800-4523-11eb-840d-6825aa43b635.png)
![image](https://user-images.githubusercontent.com/71107353/103031582-c0750b00-4523-11eb-95f6-242ec2d49c9c.png)

**Your GLPI setup (you can find it in Setup > General menu, System tab)**

[code]   GLPI 9.5.3 ( => /var/www/html) Installation mode: TARBALL
--

Operating system: Linux glpi 5.4.73-1-pve #1 SMP PVE 5.4.73-1 (Mon, 16 Nov 2020 10:52:16 +0100) x86_64 PHP 7.3.19-1~deb10u1 apache2handler (Core, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apache2handler, apc, apcu, bz2, 	calendar, ctype, curl, date, dom, exif, fileinfo, filter, ftp, gd, gettext, hash, iconv, imap, intl, json, ldap, libxml, 	mbstring, mysqli, mysqlnd, openssl, pcre, pdo_mysql, posix, readline, session, shmop, sockets, sodium, standard, sysvmsg, 	sysvsem, sysvshm, tokenizer, wddx, xml, xmlreader, xmlrpc, xmlwriter, xsl, zip, zlib) Setup: max_execution_time="600" memory_limit="64M" post_max_size="8M" safe_mode="" session.save_handler="files" 	upload_max_filesize="20M"  Software: Apache/2.4.38 (Debian) (Apache/2.4.38 (Debian) Server at helpdesk.yourtekpro.com Port 80) 	Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:84.0) Gecko/20100101 Firefox/84.0 Server Software: Debian 10 	Server Version: 10.3.27-MariaDB-0+deb10u1 	Server SQL Mode:  	Parameters: glbi@127.0.0.1/glbi 	Host info: 127.0.0.1 via TCP/IP 	 PHP version is at least 7.2.0 - Perfect! Sessions support is available - Perfect! Allocated memory > 64 Mio - Perfect! mysqli extension is installed ctype extension is installed fileinfo extension is installed json extension is installed mbstring extension is installed iconv extension is installed zlib extension is installed curl extension is installed gd extension is installed simplexml extension is installed intl extension is installed ldap extension is installed apcu extension is installed Zend OPcache extension is installed xmlrpc extension is installed CAS extension is installed exif extension is installed zip extension is installed bz2 extension is installed sodium extension is installed Database version seems correct (10.3.27) - Perfect! Timezones seems loaded in database The log file has been created successfully. Write access to /var/www/html/config has been validated. Write access to /var/www/html/files has been validated. Write access to /var/www/html/files/_dumps has been validated. Write access to /var/www/html/files/_sessions has been validated. Write access to /var/www/html/files/_cron has been validated. Write access to /var/www/html/files/_graphs has been validated. Write access to /var/www/html/files/_lock has been validated. Write access to /var/www/html/files/_plugins has been validated. Write access to /var/www/html/files/_tmp has been validated. Write access to /var/www/html/files/_cache has been validated. Write access to /var/www/html/files/_rss has been validated. Write access to /var/www/html/files/_uploads has been validated. Write access to /var/www/html/files/_pictures has been validated. Write access to /var/www/html/marketplace has been validated. Web access to the files directory should not be allowed but this cannot be checked automatically on this instance. Make sure access to error log file (/files/_log/php-errors.log) is forbidden; otherwise review .htaccess file and web server configuration.

GLPI_ROOT: /var/www/html GLPI_CONFIG_DIR: /var/www/html/config GLPI_VAR_DIR: /var/www/html/files GLPI_MARKETPLACE_DIR: /var/www/html/marketplace GLPI_USE_CSRF_CHECK: 1 GLPI_CSRF_EXPIRES: 7200 GLPI_CSRF_MAX_TOKENS: 100 GLPI_USE_IDOR_CHECK: 1 GLPI_IDOR_EXPIRES: 7200 GLPI_TELEMETRY_URI: https://telemetry.glpi-project.org GLPI_INSTALL_MODE: TARBALL GLPI_NETWORK_MAIL: glpi@teclib.com GLPI_NETWORK_SERVICES: https://services.glpi-network.com GLPI_MARKETPLACE_PRERELEASES:  GLPI_USER_AGENT_EXTRA_COMMENTS:  GLPI_AJAX_DASHBOARD: 1 GLPI_CALDAV_IMPORT_STATE: 0 GLPI_DEMO_MODE: 0 GLPI_FORCE_EMPTY_SQL_MODE: 1 GLPI_DOC_DIR: /var/www/html/files GLPI_CACHE_DIR: /var/www/html/files/_cache GLPI_CRON_DIR: /var/www/html/files/_cron GLPI_DUMP_DIR: /var/www/html/files/_dumps GLPI_GRAPH_DIR: /var/www/html/files/_graphs GLPI_LOCAL_I18N_DIR: /var/www/html/files/_locales GLPI_LOCK_DIR: /var/www/html/files/_lock GLPI_LOG_DIR: /var/www/html/files/_log GLPI_PICTURE_DIR: /var/www/html/files/_pictures GLPI_PLUGIN_DOC_DIR: /var/www/html/files/_plugins GLPI_RSS_DIR: /var/www/html/files/_rss GLPI_SESSION_DIR: /var/www/html/files/_sessions GLPI_TMP_DIR: /var/www/html/files/_tmp GLPI_UPLOAD_DIR: /var/www/html/files/_uploads GLPI_NETWORK_REGISTRATION_API_URL: https://services.glpi-network.com/api/registration/ GLPI_MARKETPLACE_PLUGINS_API_URI: https://services.glpi-network.com/api/glpi-plugins/ GLPI_I18N_DIR: /var/www/html/locales GLPI_VERSION: 9.5.3 GLPI_SCHEMA_VERSION: 9.5.3 GLPI_MIN_PHP: 7.2.0 GLPI_YEAR: 2020

htmlawed/htmlawed version 1.2.5 in (/var/www/html/vendor/htmlawed/htmlawed) phpmailer/phpmailer version 6.1.6 in (/var/www/html/vendor/phpmailer/phpmailer/src) simplepie/simplepie version 1.5.6 in (/var/www/html/vendor/simplepie/simplepie/library) tecnickcom/tcpdf version 6.3.5 in (/var/www/html/vendor/tecnickcom/tcpdf) michelf/php-markdown in (/var/www/html/vendor/michelf/php-markdown/Michelf) true/punycode in (/var/www/html/vendor/true/punycode/src) iamcal/lib_autolink in (/var/www/html/vendor/iamcal/lib_autolink) sabre/dav in (/var/www/html/vendor/sabre/dav/lib/DAV) sabre/http in (/var/www/html/vendor/sabre/http/lib) sabre/uri in (/var/www/html/vendor/sabre/uri/lib) sabre/vobject in (/var/www/html/vendor/sabre/vobject/lib) laminas/laminas-cache in (/var/www/html/vendor/laminas/laminas-cache/src) laminas/laminas-i18n in (/var/www/html/vendor/laminas/laminas-i18n/src) laminas/laminas-serializer in (/var/www/html/vendor/laminas/laminas-serializer/src) monolog/monolog in (/var/www/html/vendor/monolog/monolog/src/Monolog) sebastian/diff in (/var/www/html/vendor/sebastian/diff/src) elvanto/litemoji in (/var/www/html/vendor/elvanto/litemoji/src) symfony/console in (/var/www/html/vendor/symfony/console) scssphp/scssphp in (/var/www/html/vendor/scssphp/scssphp/src) laminas/laminas-mail in (/var/www/html/vendor/laminas/laminas-mail/src/Protocol) laminas/laminas-mime in (/var/www/html/vendor/laminas/laminas-mime/src) rlanvin/php-rrule in (/var/www/html/vendor/rlanvin/php-rrule/src) blueimp/jquery-file-upload in (/var/www/html/vendor/blueimp/jquery-file-upload/server/php) ramsey/uuid in (/var/www/html/vendor/ramsey/uuid/src) psr/log in (/var/www/html/vendor/psr/log/Psr/Log) psr/simple-cache in (/var/www/html/vendor/psr/simple-cache/src) mexitek/phpcolors in (/var/www/html/vendor/mexitek/phpcolors/src/Mexitek/PHPColors) guzzlehttp/guzzle in (/var/www/html/vendor/guzzlehttp/guzzle/src) guzzlehttp/psr7 in (/var/www/html/vendor/guzzlehttp/psr7/src) wapmorgan/unified-archive in (/var/www/html/vendor/wapmorgan/unified-archive/src) paragonie/sodium_compat in (/var/www/html/vendor/paragonie/sodium_compat/src) phpCas version 1.3.6 in (/usr/share/php/CAS/source)

**Comment by mrjd19811** [comment]
More information, when the issue happens it goes sequentially from a few hundred tickets ago.  During my testing I have gone from 2550 to 2562 when it fails, which is more often than not. My next true ticket number should bein the 2700's

**Comment by mrjd19811** [comment]
More information, this could be related with mailgate being stuck?  It seems to have resolved itself after mailgate was unstuck, but why is email flowing if mailgate is stuff? Will update if more is found.

**Comment by mrjd19811** [comment]
I'm leaning towards this being an issue while mailgate is running and or stuck. Yesterday our mailgate was stuck running from 1pm till almost 7pm, during which it reliably ate the emails in the inbox, assigned it a ticket number from the past, 160 or so less than current. in sequence. (2570 instead of 2700's) 
It did not bind it as a follow up, nor did it create a new ticket with this number that we can tell.

After discovering the mailgate locked up we added the unlock cronjob. And ran some tests,and everything worked as expected.
This morning however it did happen again, but only once so far. I am wondering if this one time was during a mailgate run, and the new mail came in during its run, or if it locked up again and the email came in before the cronjob reset it.

**Comment by kridlak** [comment]
> After discovering the mailgate locked up we added the unlock cronjob.


Hello, we have the same bug.
Can you say what command are using in unlock cronjob? Thank you.

**Comment by mrjd19811** [comment]
> > After discovering the mailgate locked up we added the unlock cronjob.
> 
> Hello, we have the same bug.
> Can you say what command are using in unlock cronjob? Thank you.

Sure, it's part of the glpi console program, php bin/console task:unlock

I use */11 * * * * php /var/www/html/bin/console task:unlock

Keep in mind this does not fix the issue. But it did improve our frequency of it happening. We used 11min to offset in case it tries to kill a legitimate wait on our 5min repeating mailgate

**Comment by tomolimo** [comment]
Hello @kridlak 
Could you give the list of plugins that are installed/enabled in your GLPI system?
Thank you
Regards,
Tomolimo

**Comment by kridlak** [comment]
Hello, @tomolimo 
This is the list:
1. Form Creator v. 2.10.4
2. IP Adressing v. 2.9.0.
3. OCS Inventory NG v. 1.7.3
4. Reports v.1.14
5. Behaviours v.2.4.1

**Comment by Terrokkar** [comment]
Hello all,

Have you maybe found some solution to this? Its really tedious to doublecheck all the tickets are created properly and very annoying to spam users with bogus mails.
